src/ray/raylet/node_manager.cc

```diff
@@ -854,12 +854,19 @@ void NodeManager::ProcessRegisterClientRequestMessage(
   } else {
     // Register the new driver.
     const WorkerID driver_id = from_flatbuf<WorkerID>(*message->worker_id());
+    const JobID job_id = from_flatbuf<JobID>(*message->job_id());
     // Compute a dummy driver task id from a given driver.
     const TaskID driver_task_id = TaskID::ComputeDriverTaskId(driver_id);
     worker->AssignTaskId(driver_task_id);
-    worker->AssignJobId(from_flatbuf<JobID>(*message->job_id()));
+    worker->AssignJobId(job_id);
     worker_pool_.RegisterDriver(std::move(worker));
     local_queues_.AddDriverTaskId(driver_task_id);
+    RAY_CHECK_OK(
+      gcs_client_->job_table().AppendJobData(JobID(driver_id),
```

Comment:
I think before this is merged, we should first merge https://github.com/ray-project/ray/pull/5099. At the moment, these definitions are pretty inconsistent.

Reply:
This will get nicer once https://github.com/ray-project/ray/pull/5110 is merged.